crates/iceberg/src/spec/table_metadata.rs

```diff
@@ -278,6 +330,229 @@ impl TableMetadata {
         self.snapshots
             .insert(snapshot.snapshot_id(), Arc::new(snapshot));
     }
+
+    /// Normalize this partition spec.
+    ///
+    /// This is an internal method
+    /// meant to be called after constructing table metadata from untrusted sources.
+    /// We run this method after json deserialization.
+    /// All constructors for `TableMetadata` which are part of `iceberg-rust`
+    /// should return normalized `TableMetadata`.
+    ///
+    /// It does:
+    /// * Validate the current schema is set and valid
```

Comment:
Those comments are nice. Can we improve them by moving to correcting functions? It's possible for us to refactor or change parts of them, which may make the comments easier to be invalid.